validator/client/src/main/java/tech/pegasys/teku/validator/client/ValidatorClientService.java

```diff
@@ -87,16 +88,19 @@ public static ValidatorClientService create(
     ValidatorClientService validatorClientService =
         new ValidatorClientService(eventChannels, beaconNodeApi, forkProvider);
 
-    asyncRunner
-        .runAsync(
+    SafeFuture<Void> future =
+        asyncRunner.runAsync(
             () ->
                 validatorClientService.initializeValidators(
-                    config, validatorApiChannel, asyncRunner, services))
-        .finish(err -> LOG.error("Unable to initialize validators", err));
-
+                    config, validatorApiChannel, asyncRunner, services));
+    validatorClientService.setInitializeValidatorsFuture(future);
```

Comment:
Turns out since this is all within the one class we don't need the setter.  So this can just be:

```
asyncRunner.runAsync(
           () -> validatorClientService.initializeValidators(
                      config, validatorApiChannel, asyncRunner, services))
        .propagateTo(validatorClientService. initializationComplete);
```

And remove the call to `initializationComplete.complete` inside `initializeValidators`.